### PR TITLE
FE_Hermite objects for solving with Hermite finite elements

### PR DESCRIPTION
**Pull request by ivweber**
This is an initial implementation of Hermite finite elements capable of generating a finite element object with the correct continuity enforced across element boundaries. It is currently capable of solving some problems, but is not compatible with various boundary methods in the VectorTools namespace. This will be addressed in a later pull request.

### CODE REVIEW

**Comment by bangerth** [comment]
@ivweber Thanks for the patch. We are currently all busy going through the (lengthy) task list to get the next release out, but someone will take a look at your patch once that is finished.

If, in the meantime, you want to work on this, try whether you can merge ("squash") the many commits you have into one commit.

**Comment by ivweber** [comment]
@bangerth Thank you for letting me know, I've squashed the commits together as requested.

**Review by peterrum — APPROVED** [review]
I am in favor of merging this PR as it is (after it has been rebased and a change-log entry has been added). This PR has been open for too long and the content has been reviewed in https://github.com/dealii/dealii/pull/13239. Thanks @ivweber!

**Comment by ivweber** [comment]
@peterrum I have just rebased onto the current deal.II master, and added a change-log file under major changes.

**Comment by peterrum** [comment]
> conversion turns floating-point number into integer: 'typename std::enable_if<std::is_integral<unsigned int>::value, double>::type' (aka 'double') to 'unsigned int' [-Werror,-Wfloat-conversion]
    return (fe_degree == 0) ? 0 : std::floor((fe_degree - 1) / 2);


You need to write ` return (fe_degree == 0) ? 0 : static_cast<int>(std::floor((fe_degree - 1) / 2));`

File: source/fe/fe_hermite.cc [hunk 1]
```diff
@@ -0,0 +1,996 @@
...
+DEAL_II_NAMESPACE_OPEN
+
+
+
+namespace internal
+{
+  inline unsigned int
+  get_regularity_from_degree(const unsigned int fe_degree)
+  {
+    return (fe_degree == 0) ? 0 : std::floor((fe_degree - 1) / 2);
```
Comment (masterleinad) [inline source/fe/fe_hermite.cc:68]:
```suggestion
    return (fe_degree == 0) ? 0 : (fe_degree - 1) / 2;
```

**Comment by peterrum** [comment]
@ivweber For some reason, the github actions stuck again. I have rebased and forced pushed again. I will merge as soon as the tests have passed! Thanks a lot! Hope that GitHub won't cause the same issues in your next PR!

**Review by peterrum — APPROVED** [review]
I am merging, since the error is unrelated to this PR. It is already tracked in https://github.com/dealii/dealii/issues/14693.

**Comment by marcfehling** [comment]
The oneAPI CI complains about deprecated implicit conversions:
```
/home/runner/work/dealii/dealii/source/fe/fe_hermite.cc:494:34: error: 'ComponentMask' is deprecated: Implicit conversions from std::vector<bool> to ComponentMask are deprecated! [-Werror,-Wdeprecated-declarations]
                                 std::vector<bool>(1, true)))
                                 ^
/home/runner/work/dealii/dealii/include/deal.II/fe/component_mask.h:271:23: note: 'ComponentMask' has been explicitly marked deprecated here
inline ComponentMask::ComponentMask(const std::vector<bool> &component_mask)
                      ^
```

**Pull request merged**